### PR TITLE
use specProvider stateful constants in API dataProviders and Rest interfaces

### PR DESCRIPTION
**Pull request by rolfyone**
Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/provider/src/main/java/tech/pegasys/teku/api/blockselector/BlockSelectorFactory.java [hunk 1]
```diff
@@ -72,7 +75,9 @@ public BlockSelector finalizedSelector() {
   }
 
   public BlockSelector genesisSelector() {
-    return () -> optionalToList(client.getBlockAtSlotExact(UInt64.valueOf(Constants.GENESIS_SLOT)));
+    final UInt64 genesisSlot =
+        UInt64.valueOf(specProvider.getGenesisSpec().getConstants().getGenesisSlot());
```
Comment (mbaxter) [inline data/provider/src/main/java/tech/pegasys/teku/api/blockselector/BlockSelectorFactory.java]: (optional) We could update the genesis slot / epoch to type `UInt64`

File: data/serializer/src/test/java/tech/pegasys/teku/api/schema/BeaconValidatorsTest.java [hunk 2]
```diff
@@ -21,20 +21,26 @@
...
+import tech.pegasys.teku.spec.StubSpecProvider;
+import tech.pegasys.teku.spec.util.BeaconStateUtil;
 import tech.pegasys.teku.ssz.SSZTypes.SSZList;
 import tech.pegasys.teku.util.config.Constants;
 
 class BeaconValidatorsTest {
   private final DataStructureUtil dataStructureUtil = new DataStructureUtil();
+  private final BeaconStateUtil beaconStateUtil =
+      StubSpecProvider.create().getGenesisSpec().getBeaconStateUtil();
+  private final SpecProvider specProvider = StubSpecProvider.create();
```
Comment (mbaxter) [inline data/serializer/src/test/java/tech/pegasys/teku/api/schema/BeaconValidatorsTest.java]: We should just create the `SpecProvider` once:
```suggestion
  private final SpecProvider specProvider = StubSpecProvider.create();
  private final BeaconStateUtil beaconStateUtil =
      specProvider.getGenesisSpec().getBeaconStateUtil();
```

**Review by mbaxter — APPROVED** [review]
LGTM

File: data/serializer/src/test/java/tech/pegasys/teku/api/schema/BeaconValidatorsTest.java (hunk 2 above)
Comment (rolfyone) [reply data/serializer/src/test/java/tech/pegasys/teku/api/schema/BeaconValidatorsTest.java]: hrm i wonder how i did that, the whole point was to get the beaconStateUtil from provider :/ good catch

**Pull request merged**